lib/components/map/connected-transit-vehicle-overlay.js

```diff
@@ -124,7 +124,7 @@ const mapStateToProps = (state) => {
   if (viewedRoute?.routeId) {
     vehicleList = route?.vehicles?.map((vehicle) => {
       vehicle.routeType = route?.mode
-      vehicle.routeColor = route?.color
+      vehicle.routeColor = route?.color ? '#' + route.color : null
```

Comment:
```suggestion
      vehicle.routeColor = !route?.color.includes("#") ? '#' + route.color : route?.color
```

This covers more cases while (hopefully) still fixing the issue

Reply:
Good points! Added the change to https://github.com/opentripplanner/otp-react-redux/pull/807/commits/efea50e67e2786c99d8057c566fb081feb9a6b07. Also handled the case where the color is undefined, null or "" by defaulting it to a dark grey.
![Pasted Graphic 66](https://user-images.githubusercontent.com/62163307/222800727-52073e85-0beb-43dc-b588-938da95530be.png)